compiler/pipes/convert-sprintf-calls.cpp

```diff
@@ -0,0 +1,179 @@
+// Compiler for PHP (aka KPHP)
+// Copyright (c) 2021 LLC «V Kontakte»
+// Distributed under the GPL v3 License, see LICENSE.notice.txt
+
+#include "compiler/pipes/convert-sprintf-calls.h"
+
+#include <compiler/gentree.h>
+
+struct FormatCallInfo {
+  FormatCallInfo() : args(VertexRange(Vertex::iterator{}, Vertex::iterator{})) {}
+  // stores a variable that contains all the call arguments,
+  // if all arguments are constant
+  VertexAdaptor<op_var> var;
+  // stores an array that contains all the call arguments,
+  // if at least one argument is a function call or a variable
+  VertexAdaptor<op_array> array;
+  // flag for the presence of a value in format_args_var
+  bool is_var{false};
+  // stores a set of vertices of the call arguments from var or array
+  VertexRange args;
+};
+
+struct FormatPart {
+  enum class FormatSpecType {
+    Decimal,
+    String
```

Comment:
We don't name enum values from a capital letter :)

Reply:
Fixed.